renameAsSudo: run command as an array, not a string

mistydemeo opened the pull request:
This ensures that this has no chance of failing if spaces or quotes are in command names, since shell splitting won't take place.

Commit:
Add new code review guide

Commit:
Introduce remaining sections of guide

Commit:
Link to Github's fork guide

Commit:
Link initial Github mention

Commit:
Clarify line comments belong in the diff, not commit pages

Commit:
Document how to use interactive rebase

Commit:
Fix ordered list

Commit:
When to submit: fix minor typo

mcantelon committed:
Created granular AtoM DIP upload config form.

Replaced the AtoM DIP upload config form, which specified
a command and arguments, with a granular form with fields
for different DIP upload options.

Commit:
Use form to construct AtoM upload job arguments

Commit:
Add stub for viewing levels of description

Conflicts:
	src/dashboard/src/components/administration/urls.py

Commit:
Add API to fetch JSON levels of description

Commit:
ingest_upload_destination_url_check: use new URL setting

Commit:
Add new LevelofDescription model

This class is used to track levels of description from an access
system (e.g. AtoM), and is used to tag directories and files when
producing a SIP in order to correctly map items into a DIP.

refs #6980

Commit:
Add helper to fetch levels of description from AtoM

Commit:
Add API method to fetch levels of description from AtoM

Commit:
Add FileLevelOfDescription class

This tracks the level of description for a filesystem entry in a SIP,
either a file or a directory.

mcantelon committed:
Added stub for SIP arrange LOD editing.

Added a button, and stumped in logic, for editing a directory's
level of description.

Commit:
createMETS2: Write arranged structMap to METS, if present

This hierarchical structMap is created by assigning levels of description
to a SIP during SIP arrangement.

Commit:
Add API to store levels of description for SIP objects

Commit:
Correct name of view to fetch levels of description from AtoM

mcantelon committed:
Added AtoM level of description admin UI.

mcantelon committed:
Added level of description edit modal.

Added a pop-up form in the SIP arrange interface that allows editing of
a directory's metadata.

mcantelon committed:
Allow files to have levels of description.

mcantelon committed:
Assign SIP UUID to directory levels of desc.

Added logic, during SIP creation from the arrange panel, to assign
the SIP UUID to each DB entry for a directory's level of description.

mcantelon committed:
Remove restriction of assigning LOD to files.

Removed the UI restriction preventing assigning level of description,
in the SIP arrange panel, to files.

mcantelon committed:
Changed file level of descriptions storage.

Changed how file level of descriptions are stored so once a
SIP is created the relative location becomes relative to the
SIP directory (rather than, before it becomes a SIP, the
arrange directory).

Hwesta committed:
Fix missing structMap, TYPE attrs

refs #7068, #7071

Remove the FileLevelOfDescription and replace it with attributes on
SIPArrange. Instead of attaching the level of description to the
FileLevelOfDescription, populate the new level_of_description field in
SIPArrange. This is easier to find, set and update when the SIP is created,
and less error prone when creating a SIP from not top level directories. Do
not delete SIPArrange directories, as they are now needed for level of
description information.

In createMETS2, search based on the path relative to the objects directory.
Query SIPArrange instead of FileLevelOfDescription.

Hwesta committed:
Add update SQL for main_siparrange

refs #7068

Add SQL to update SIPArrange for new columns. syncdb will create these
on a fresh install, but not on update.

Hwesta committed:
Add SIP label to objects div

refs #7068

Add the label from the directory the SIP was created from to the
logical structMap's objects div. If there was an objects directory in
the arranged SIP already, delete it. Delete TYPE=Directory from the SIP
level div.

Hwesta committed:
LoD: Cleanup fetch & display of levels of description

refs #7070

Fix get_atom_levels_of_description to work correctly with clear=False.
New LoD get unique sortorders, starting at the max value of the
existing sortorders, since swapping sortorder values will not work if
they have the same value.

Fix hiding of promote/demote arrows only when sortorder is the max or
min value (instead of 1 or length, since values may be deleted). Align
all demote arrows vertically. Use STATIC_URL instead of hardcoded value.

Clean up _atom_levels_of_description_sort_adjust remove unnecessary
iterate and inefficient swap.  Misc other minor fixes.

mcantelon committed:
Directory AJAX feed provides directory metadata.

Updated the AJAX feed for providing SIP arrange directory data
with level of directory metadata.

mcantelon committed:
Display level of description in SIP arrange.

Updated client-side UI logic to display level of description for
directories.

mcantelon committed:
Re-enabled editing of file metadata.

mcantelon committed:
Enforce valid user interaction in arrange panel.

Added disabling of SIP arrange panel action buttons when the user has
highlighted an item that the action should not apply to (creating
directories within files, for example).

Added an error alert in the SIP arrange UI if someone tries to
create a directory within a file.

Commented and cleaned up JavaScript.

Conflicts:
	src/dashboard/src/media/js/ingest/ingest_file_browser.js

mcantelon committed:
Minor CSS tweak to SIP arrange paging links.

mcantelon committed:
Auto-search transfer backlog in originals panel.

If a open opens the originals file browser and the transfer backlog isn't
yet searched, trigger a search.

Hwesta committed:
Templates: Remove load url from future

refs #7305

Not needed in Django 1.5, deprecated in 1.7

Hwesta committed:
ATK: Replace django's simplejson with json

refs #7305

Deprecated in Django 1.7, and builtin json module is equivalent.

Hwesta committed:
Dashboard: Update mimetype to content_type in HttpResponse

refs #7305

mimetype is deprecated in Django 1.7, replaced by content_type

Hwesta committed:
ATK, AS: Update Form to take advantage of ModelForm

Use ModelForm's ability to autogenerate fields from the Model, instead
of redefining fields. Move labels and choices to the model.

Hwesta committed:
Forms: Add fields listing to all ModelForms

refs #7305

Django 1.7 no longer defaults to fields being everything because that
is a security risk. Add explicit specification of fields to all
ModelForms.

Hwesta committed:
Update WSGI, manage.py for Djange 1.7

refs #7305

Hwesta committed:
Models: Set defaults on BooleanFields

refs #7305

Hwesta committed:
Django: Update to 1.7

refs #7305

Update Django and Tastypie versions. Update test runner default to
suppress warnings. Update all client scripts to initialize Django
before using the ORM.

Commit:
renameAsSudo: run command as an array, not a string

This ensures that this has no chance of failing if spaces or quotes
are in command names, since shell splitting won't take place.

Hwesta committed:
Dashboard: Misc updates reducing warnings

get_query_set deprecated in favor of get_queryset. cycle now takes
variables and literals must be quoted.

Hwesta committed:
Postinst: Upgrade pip dependencies, handle pyo's

refs #7305

Pip does not remove pyo's that were generated after install, so remove
those before upgrading Django as it causes a dirty upgrade. Change pip
command to update dependencies.

Hwesta committed:
URLs: Update to new URL syntax

refs #7305

All urls created with the url() function.  Patterns will be going away
in 1.8, so standardize how it's imported and used to make removing it
easier later.

Hwesta committed:
Common tests: Fix test

The DB enforces that events must point to a valid file UUID.

Commit:
dev-helper: remove restart services

This also exists in the archivematica-devtools repository; removing
this extra copy to avoid maintaining two versions.

mistydemeo committed:
WIP CONTRIBUTING.md

Hwesta committed:
Dependencies: Improve SSL dependencies

Add better openSSL support. See
https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning

Hwesta committed:
ArchivesSpace: Do not open matching GUI with JS

Hwesta committed:
DIP: Add DIP generation helper skeleton

Helper to generate information needed by the DIP that is only available in
the SIP. E.g. parsing metadata files.

Hwesta committed:
ArchivesSpace: Parse archivesspaceids.csv for filename->resource ID

Hwesta committed:
ArchivesSpace: Minor fixes

Hwesta committed:
ArchivesSpace: Add query by field function

Add function to return a resource based on an exact query by a specific field.

Hwesta committed:
Add helper to collect all metadata files

Helper searches transfer and SIP metadata directories for all copies of a
file with the provided filename.

Hwesta committed:
Cleanup fetch_levels_of_description_from_atom

Squash with Add API method to fetch levels of description?

Hwesta committed:
ArchivesSpace: Provide access file size & format info

refs #8696

When uploading to ArchivesSpace, also upload the access file's size and
format name.  file_format_name is a restricted vocabulary and should be
configured in ArchivesSpace first so upload succeeds.

Hwesta committed:
ArchivesSpace: add subjects & agents to children

refs #8697

Hwesta committed:
JS: Remove unnecessary monkey patch

Commit:
ArchivesSpace, ATK: add identifier search

refs #8695

Commit:
Pair matcher: add identifier search

refs #8695

Hwesta committed:
Dashboard: AS/ATK cleanup

Remove unused function, use messages for form confirmation, update ATK config
saving to match AS, format template.

Hwesta committed:
Remove vendored requests 1.2.0

Hwesta committed:
Remove vendored requests library, list in requirements

Hwesta committed:
Tests: Fix FPRclient tests

Hwesta committed:
FPRClient: Reduce calls

Append / to prevent 301 MOVED PERMANENTLY responses.

Hwesta committed:
Update fpr-admin commit

Commit:
CONTRIBUTING: link to PEP 257

Hwesta committed:
Remove suggestion on bikeshedding issue

Hwesta committed:
Minor contributing.md additions

Hwesta committed:
CONTRIBUTING: section on tests

jhsimpson committed:
change sql statements to be case sensitive

refs #6267

original fix for this issue applied to the
matching gui, this fix is for the
upload-archivistsToolkit.py client script.
Updated sql statements to be case sensitive
and added an empty value for label column
of DigitalObjects table, because in some
versions of ATK, that column is not nullable.

Commit:
ArchivesSpaceClient: add method to return resources by accession

Commit:
ArchivistsToolkitClient: update find_collection_ids documentation

Commit:
ArchivesSpaceClient: document find_collections

This has diverged in behaviour from its ArchivistsToolkitClient
counterpart.

Commit:
Add initial /access/ APIs

Commit:
Elasticsearch: update transferfile index

These new fields are used by the new appraisal tab.

Commit:
Elasticsearch: Index new transferfile fields

Commit:
Ingest: add API for appraisal tab interface

This new data format is more detailed than the old one, and includes
many metadata properties which are required for the appraisal tab.

This doesn’t yet replace the old one, as both UIs will be supported
for a period.

Commit:
Include appraisal tab as submodule

Commit:
AS/ATK clients: return "has_children" key in results

Commit:
/ingest/backlog/: don’t redirect to self

If no query is provided, use a MATCH_ALL_QUERY instead.

Commit:
elasticSearchFunctions: add functions to get and set file tags

Commit:
Add new /file/ API

This API is focused on retrieval of file metadata; it is currently
used by the appraisal tab.

Supports:

* Retrieval of file record data
* Add/list/remove file tags
* Retrieve file's bulk_extractor reports

hakamine committed:
Initialize Django in MCP client scripts ref. #8879

Standalone scripts require to initialize Django to avoid
an AppRegistryNotReady exception

So far identified 3 scripts, others may also need to be fixed.

Commit:
filesystem_ajax: add API to download file via UUID

Commit:
arrange_contents: fix calling without the `path` parameter

DEFAULT_ARRANGE_PATH is not base64-encoded; trying to decode it raises
an exception.

Commit:
arrange_contents: handle non-base64 values

Commit:
/filesystem/delete/arrange: make filepath mandatory

Defaulting to an empty string means that this deletes the entire
contents of arrange. This is very easy to do by accident when calling
this API manually, for example if the user accidentally passes the
parameters as URL parameters or JSON data instead of form-encoded data.

Commit:
/filesystem/delete: make filepath mandatory

Commit:
AS/ATK clients: add method to get levels of description

Hwesta commented on src/archivematicaCommon/lib/fileOperations.py:
```diff
@@ -116,9 +116,7 @@ def writeToFile(output, fileName, writeWhite=False):
-    command = "sudo mv \"" + source + "\"   \"" + destination + "\""
-    if isinstance(command, unicode):
-        command = command.encode("utf-8")
+    command = ["sudo", "mv", source, destination]
```
Does this still need to be encoded?

mistydemeo replied:
Testing, no - mixed encodings are fine if no shell splitting is taking place. e.g. this is valid, with one unistring and one bytestring that's not valid Unicode:

``` python
subprocess.check_output(['echo', u'é', '\xFF'])
```

Hwesta commented:
:+1:

sevein committed:
Rename endpoint, s/ransfer/transfer

Commit:
Correcting duplicate files checksums verification.

This commit changes the behaviour of the transfer checksums verification to
check correctly duplicate files (same content, different file names).

The microservice Verify transfer checksums (Job: Verify metadata directory
checksums) does not work properly in case there are same files with different
names in the transfer. For example (hash is md5 but it is the same for sha1 and sha256):

89808652ad47f2bb97df733e8a2e14ba objects/collection/folder/a.jpg
89808652ad47f2bb97df733e8a2e14ba objects/collection/folder/b.jpg

Such transfer fails and the a.jpg file is reported as corrupted (hash does not
match) because the md5deep is called

md5deep -r -n -m

which compares the computed hashes first (so the computed hash for the b.jpg
matches the line for a.jpg).

This behaviour catches duplicate files which is probably not the goal of the
Verify transfer checksums microservice.

This request call for the <hash>deep should be

md5deep -r -x

This slight change in the call works as intended: reports only those files not
listed in checksum.<hash> or those with corrupted (mismatched) hashes.
Duplicate files are matched correctly (and not reported).

Commit:
ATK/AS clients: fix return type documentation

Commit:
Access: add /levels/ API

Commit:
AS: add new API to add a child to an existing resource

Commit:
/access/: Add API to edit records

Commit:
/access/: make GET record failure consistent

Commit:
copy_to_arrange: "objects" shouldn't be dragged onto arrange root

Fixes #8957

Commit:
DIP generation helper: search in fullrecord, not identifier

Commit:
AS.find_by_field: fix returning resource_components

Commit:
AS.find_by_field: only return resources or archival_objects

Hwesta committed:
Upload-ArchivesSpace: Only send ArchivesSpace formats

ArchivesSpace only accepts a set list of file formats. Attempt to match
between FPR formats and the ones ArchivesSpace accepts, or send no format
information.

Commit:
AS client: update tests

Commit:
MCPClient: update test_parse_to_db test for new URL

Commit:
Stream files from Storage Service instead of redirecting

Fixes #8963.

Commit:
Ingest: fix marking logs/metadata not draggable

Commit:
Decouple renderBacklogSearchForm from originals_browser

Note that the openInNewTab parameter hasn't done anything since d5b83fa6.

Commit:
ingest/urls: fix typo

Commit:
Fix rendering empty responses

Commit:
Backlog search: allow custom error handling

Commit:
/access/: fix module name

Commit:
/access/: prioritize /levels over record

Commit:
/access/: fix fetching record children

Commit:
AS client: fix edit_record

Commit:
Add SIPArrangeAccessMappings table

This allows mapping SIPs under arrangement to records in a remote
archival management system.

"arrange_path" is intended to refer to the parent directory of a SIP;
other files within a SIP are implicitly assumed to exist within that
descriptive record.

Commit:
Add APIs to associate arranged SIPs with access system records

Commit:
filesystem_ajax: return proper exit codes on error

Hwesta committed:
Test: fix filesystem_ajax test

Hwesta committed:
ArchivesSpace: find_by_field uses new endpoint

find_by_field renamed to find_by_id and uses new ArchivesSpace plugin
endpoint. This should return no more than 1 result, and no false positives.

Commit:
Backlog tab: Add backlog search page with transfer table

Hwesta committed:
file_to_folder: init Django

Hwesta committed:
Models: point FPCommandOutput at legacy table

Hwesta committed:
Logging: silence verbose libraries in client script logs

Convert get_script_logger to use a dict for config. Keep formatters/handler
and loggers for passed-in root ('archivematica') and script logger.  Add a
root logger that only logs WARNING and higher from other libraries.

Notably this silences elasticsearch's verbose INFO logs.

Commit:
SIP Arrange: handle storage_service.ResourceNotFound

jhsimpson committed:
refactor notes inserts

refs #8910

When inserting notes into an Archivists Toolkit database, sometimes
the eadIngestProblem column is marked as not nullable, so pass an empty value

Also combined 3 insert statements together to re-use the same sql statement

Commit:
/contents/arrange/: return 404 if directory is missing

jhsimpson committed:
change uuid used in determing file format

the access file does not have file identification info in the database
use the original file uuid instead.

Hwesta committed:
Logging: Configure dashboard

Hwesta committed:
Logging: dashboard debug logfile

Hwesta committed:
Logging: configure MCPClient

Hwesta committed:
Logging: cleanup client scripts config

Hwesta committed:
Logging: cleanup MCPServer config

Commit:
Backlog tab: Add ES document and code to store backlog transfer information

Hwesta committed:
Filesystem AJAX: Remove unused 'delete' endpoint

/filesystem/delete/ deleted from disk, which is no longer used since SIP
arrange uses path manipulation and the storage service handles most data
transfer.

Hwesta committed:
Access API: Return arrange path with Access

When returning collections, return the mapped path from SIP Arrange. This
allows the SipArrange nodes to be children of ArchivesSpace nodes, since all
their ancestors have paths too.

Hwesta committed:
Filesystem API: Don't return error if no SIP arrange paths

Distinguish no entries found because directory doesn't exist from no entries
found because we're querying the default directory and nothing has been
created.

Hwesta committed:
Access API: Return directory path on creation

Commit:
Backlog tab: Integrate table with datatables.js and add ajax pagination

sevein committed:
Add django-mysqlpool in mcp-{client,server}

Hwesta committed:
ArchivesSpace: Fix create child

Remove trailing , from parent definition when creating a resource component
child.  Add tests.

Hwesta committed:
ArchivesSpace: Add delete functionality

Hwesta committed:
Access API: Add delete endpoint

Delete endpoint removes associated SIPArrange mapping then deletes from
ArchivesSpace. SIPArrange also deletes access mapping if top directory
deleted.

sevein committed:
Init Django before using ORM, refs #9040

Hwesta committed:
MCPServer: Remove unused configs

Commit:
Backlog tab: Add table sorting

jhsimpson committed:
ArchivesSpace DIP Upload tweaks

refs #9093

This commit sets the publish and restrictions flags on both digital objects
and file versions.

Also, notes are copied from the parent archival object to the digital object.
These new notes are converted to type 'general note' if the original is a
note type that can't be used with a digital object.

Sometimes a DIP does not include access derivatives,
for example when the user supplies access copies
(manual normalization).  Those access copies are
listed in the Files table as 'original'

This commit sets the digital object format and size and title
based on the original in these cases.

Commit:
Backlog tab: Add backlog transfer deletion request ability

hakamine committed:
AtoM DIP upload --rsync-command fix ref. #9095

Commit:
Backlog tab: Add backlog to menu bar

Commit:
Backlog tab: Add backlog package download, move js to own file

sevein committed:
Support multiple processing configs, refs #9105

Commit:
Backlog tab: Add docstrings

Commit:
ArchivesSpace: don't let client's session expire

Commit:
ArchivesSpace: filter accession items appropriately

Commit:
Update appraisal tab commit

Commit:
Access move_within_arrange: Don't strip leading slash from filepath

Commit:
ArchivesSpace: use /tree, not /search, to determine existence of children

Commit:
ArchivesSpace: update test cassettes for client updates

Commit:
ArchivesSpaceClient: return note data

Commit:
ArchivistsToolkitClient: implement stub note data

Commit:
ArchivesSpaceClient: allow editing notes

Commit:
ArchivesSpaceClient: allow dates to be edited

Commit:
ArchivesSpace: allow specifying dates when creating children

sevein committed:
Add processing configuration API endpoint, refs #9105

Commit:
Checksum type: Initial checksum setting added to admin tab

Commit:
Checksum type: Refactor file checksum functions to be more generalized

Now allows for md5, sha1/256/512, and others supported by hashlib

Hwesta committed:
Tests: pytest ignore JS tests

Hwesta committed:
Fixtures: Add fixtures for broken foreign keys

Hwesta committed:
FPRClient: Adding entries is atomic

Django now uses transactions.  Mark adding a new FPR entry as atomic so
that it can handle exceptions and rollback. Testcase uses transactions.

Hwesta committed:
ParseMETS: ignore related event

When parsing METS for reingest, the derivation event is not used (and no
events are parsed). Remove it from the Derivation entry as the foreign key
will never be valid.

Hwesta committed:
Manage.py: Set default settings

Commit:
ArchivesSpace: add tests for record editing

Commit:
Checksum type: Add checksum type to schema, model, and funcs

Commit:
ArchivesSpaceClient: fix variable reference

Commit:
ArchivesSpaceClient: allow adding notes to new records

Commit:
Access views: allow specifying notes in new records

Commit:
Checksum type: Move get_setting helper to more generic location

get_setting is now in archivematicaFunctions

Commit:
Checksum type: updateSizeAndChecksum now uses the checksum type setting

Commit:
Checksum type: Use File's checksum type when verifying on ingest

Commit:
Backlog tab: Make file download streaming

Commit:
Backlog tab: Create a common delete_request form

Commit:
Remove unused init.d scripts

Hwesta committed:
Models: File.currentlocation can be null if deleted

Commit:
Checksum type: Add more checksum algorithms when verifying METS

Commit:
Checksum type: Add DashboardSettings row for checksum type to db

NOTE: This change requires you to run recreateDB to update the database schema

Commit:
Checksum type: Allow other checksum algorithms in verifyChecksum

Commit:
Checksum type: Add checksum type to fixity check tag

Commit:
Models: add comment to Event.linking_agent

Commit:
Archival storage: fix AIP pagination

This reworks archival storage's search view in order to fix paging of
AIP searches.

AIP searches are very broken in the current iteration. For historical
reasons, searches are performed against the aips/aipfile index, which
is limited to returning only AIP-related fields. This is done because
aips/aipfile contains relevant information which is not present in the
aips/aip index, which contains actual AIP data.

The issue arises when paging is applied. Because these are file results,
there are many files to one AIP; these are incorrectly paged as file
results. This means that each page, instead of containing ten AIPs as
intended, contains ten files' worth of AIPs - which could be a single
AIP, if the first returned AIP contains more than one file.

In addition to fetching metadata from aips/aipfile, the view later uses
the fetched UUIDs to fetch records from aips/aip - in order to have
enough information to seed the AIC form, and some other additional
metadata that aips/aipfile is missing. It currently does this by
performing one query for every fetched UUID, which is fairly wasteful.

This patch simplifies the current workflow to the following:

* Run the query against aips/aipfile, with and an aggregation which
returns all unique UUIDs in the requested set and the count of files
within that UUID. This query is set to return no results - only the
aggregation is used.
* Uses that set of UUIDs to fetch pages of AIP data from aips/aip in a
single query, which feeds the template.

Because the paging is now based on AIPs, not files within an AIP, a
page in archival storage can display the appropriate number of AIPs.
In addition, because AIPs are fetched in pages and the total number of
queries performed has been reduced substantially, performance is
substantially better.

Fixes #9030.
Fixes #9085.

Hwesta committed:
ArchivematicaFunctions: Remove unused functions, fix spacing

Hwesta committed:
Dev-helper: update pip dependencies

Hwesta committed:
Tests: Don't look for tests in share

Commit:
createPointerFile: prefix METS and PREMIS elements

sevein committed:
Processing config: small fix in form

PreconfiguredChoices was changed so config_path is passed to save() instead of
to the constructor. This commit applies its only consumer accordingly.

Hwesta committed:
Processing config: Reorder to match appearance in processing

Hwesta committed:
Unzipped Bag: Restructure doesn't move processingMCP

When verifying and restructuring a bag, do not move OPTIONAL_FILES
(currently just processingMCP).

Commit:
Elasticsearch: set type for dc:date

Fixes #8812.
Fixes #9144.

Hwesta committed:
CreateMETS2: PEP8 fixes

Commit:
models: add ArchivesSpace AIP arrangement model

Hwesta committed:
RPCServer: Clean up, fix logging

Remove duplicate logging config, replace prints with logging statements,
remove dead code.

Hwesta committed:
Tests: Use console logger

Commit:
ArchivesSpace: add APIs to add/list/edit d.o. components

Hwesta committed:
CreateMETS2: Clean up createFileSec

Remove unecessary variables populate from DB object, add comments to sections.

Hwesta committed:
ArchivematicaFunctions: find_metadata_files can only search transfers

Add option to limit search for metadata files to just the component transfers.

mistydemeo committed:
filesystem_ajax: extract core logic of copy_to_arrange into new function

mistydemeo committed:
Access: add SIP arrange features to digital object component APIs

Commit:
filesystem_ajax: extract core logic of move_within_arrange into new function

Commit:
Unify copy_to_arrange and move_within_arrange to a single endpoint

Commit:
copy_to_arrange: return an error for invalid paths

Commit:
filesystem_ajax: log to logger, not logging module

Commit:
copy_to_arrange: add comment to DEFAULT_BACKLOG_PATH use

Commit:
AS/ATK clients: include resource/resource_component type in records

Commit:
Access: provide helper to normalize record_ids

Commit:
Access: digital_object_components_files is a pure SIP arrange wrapper

sevein committed:
Add gunicorn to requirements/local.txt

It is going to be used in development environments (`gunicorn --reload`), it
may be added later to other environments.

sevein committed:
Remove MCPrpcCLI, available in devtools

https://github.com/artefactual/archivematica-devtools/blob/master/tools/mcp-rpc-cli

sevein committed:
Fix shebangs

sevein committed:
MCPClient: suspend main thread with time.sleep()

Suspend the main thread using `time.sleep()` instead of using using a lock. The
interpreter terminates the suspension when a `SIGINT` signal is received
(`CTRL+C`), giving us an opportunity to exit our threads cleanly (not done yet)
and exit the application.

Commit:
ArchivesSpaceClient: add_digital_object fixes

* The `identifier` field is mandatory.
* Remove default `identifier` handling based around `uri`.
* Don't try to add a file_version if there is no `uri`.

sevein committed:
MCPServer: Delete dead code (sqlLock stuff)

Locks are acquired in `runSQL()`, `insertAndReturnID()`, `querySQL()` and
`queryAllSQL()` (`databaseInterface`). These functions are not being used in
the MCP server anymore.

sevein committed:
MCPServer: stop using os.setuid

It can be misleading, e.g. if you are running as a different user.

sevein committed:
MCPServer: stop using PID file

Commit:
Extract ArchivesSpaceClient and ArchivistsToolkitClient into new module

hakamine committed:
Call get_script_logger() after django.setup()

For some reason get_script_logger() should be called after
django.setup() or the imports to work

sevein committed:
mysql seed: avoid too long lines

Added `--extended-insert=FALSE` and  `--complete-insert=TRUE` to mysqldump.

sevein committed:
mysql_dev.sh: take more conn parameters

hakamine committed:
Fix dashboard helpers.py typo

Commit:
access/views: update agentarchives import paths

Commit:
agentarchives: fix archivists_toolkit import path

sevein committed:
Add EnvConfigParser subclass with environment string support

Main two differences comparing it to its ancestor `SafeConfigParser` are:

- The settings are looked up first in the string environment, trying to match
  both `section` and `option` first and only `option` as a fallback, e.g.
  `ARCHIVEMATICA_SECTION_OPTION` and `ARCHIVEMATICA_OPTION`.
- All the getters (`get`, `getint`, `getfloat`, `getbool`) learn a new keyword
  argument `fallback` so its value is returned when the configuration option
  is undefined, instead of throwing exceptions like `NoSectionError` or
  `NoOptionError`, as with a Python dictionary. This mimics the behavior
  introduced in the `configparser` module in Python 3. For example, you could
  write:

      tls_enabled = config.getboolean('network', 'tls', fallback=False)

  instead of:

      try:
          tls_enabled = config.getboolean('network', 'tls')
      except (ConfigParser.NoSectionError, ConfigParser.NoOptionError):
          tls_enabled = False

Attention! This class is py2-only compatible!

sevein committed:
MCPClient: leverage upstart's service logger

Upstart logs every service (job) in a log file by the same name under
`/var/log/upstart`. Both `stdout` and `stderr` are captured. Furthermore,
Ubuntu rotates the log files as described in `/etc/logrotate.d/upstart`.
This commit takes away this responsibility from our upstart service, making
easier for Ubuntu administrators to inspect `archivematica-mcp-client` in the
standard manner.

In addition, the `pre-start` stanza that waits for Gearman to become available
is being deleted because: (1) Gearman server may not be running in the same
host, (2) MCPClient is already dealing with the situation were the Gearman
server cannot be reached (see `gearman.errors.ServerUnavailable` in
`archivematicaClient.py`) using a retry pattern with incremental delays.

sevein committed:
MCPClient: use thread-safe logger

One of the many benefits of `logging` is its inherent thread safety. Using it
saves us from having to deal with thread synchronization in order to provide our
worker threads with safe access to `stdout` and `stderr`. This commits deletes
`printOutputLock` and replaces the old print statements with calls to the
logger.

sevein committed:
MCPClient: README - clientScripts logging convention

Commit:
upload-archivistsToolkit: don't use atk module

The only thing this module was used for was obtaining a db connection
object; the function being called was a very thin wrapper around
MySQLdb.connect, so just use that instead.

refs #8850

sevein committed:
MCPClient: create logger after django.setup()

The attribute `config` in the `logging` module is not available until
`django.setup()` is invoked. This commit pospones the creation of our logger
`archivematica.mpc.client`.

mistydemeo committed:
Move getDashboardUUID into archivematicaFunctions

mistydemeo committed:
ES: specify all fields for AIP mapping

Archival storage now performs some faceting on archival storage data,
which requires those fields to exist in the mapping. Depending on
automatic mapping structure meant that these queries would throw
exceptions until the first time data was indexed containing these fields.
This was manifested in archival storage as an exception when trying to
search when the AIP store is empty.

mistydemeo committed:
Move archivematicaXMLNameSpace into archivematicaCommon, rename

mistydemeo committed:
elasticSearchFunctions: use archivematicaXMLNamespace

mistydemeo committed:
Update appraisal tab commit

mistydemeo committed:
base64: escape invalid UTF-8 characters before encoding

Fixes #9234.

mcantelon committed:
Fixed issue with passing of MySQL password to mysql_dev.

mistydemeo committed:
elasticSearchFunctions: properly handle document_id_from_field_query failure

Fixes #9256.

mistydemeo committed:
filesystem_ajax: return non-200 response on start_transfer error

refs #4757

mistydemeo committed:
Index transfer contents once

The current indexing process indexes files near the end of the
transfer process, before a storage decision has been made.
Subsequently, if the user decides to send the files to backlog, the
records must be retrieved from the index and updated with a new status.

This instead defers indexing until after a storage decision has been
made, and indexes them once with the final status.

Files which are not sent to the backlog are no longer indexed; those
records were never used.

Fixes #9265.
